File: Algorithm/Recursion & Backtracking/Subset_Recursive.cpp
#include<bits/stdc++.h>
using namespace std;

int n ; 
const int Max=1e5+5;
int arr[Max];

vector<int >temp;

void subset(int index ){
  
  if(index==n){
    for(int x:temp)
    cout<<x<<" ";
    cout<<endl;
    return ;
  }
  
  temp.push_back(arr[index]);
  subset(index+1);
  temp.pop_back();
  subset(index+1);
  
}
int main(){
  cin>>n;
  for(int i=0;i<n;i++)
  cin>>arr[i];
  
  subset(0);
  
}

File: Algorithm/Recursion & Backtracking/Subset_Recursive.py
n=int(input())
arr=list(map(int,input().split()))

temp=[]

def subset(index):
  if index==n:
    print(*temp)
    return 
  
  temp.append(arr[index]);
  subset(index+1)
  
  temp.pop()
  subset(index+1)
  



subset(0)